workers/unity/Packages/io.improbable.gdk.mobile/Editor/iOSUtils.cs

```diff
@@ -80,37 +81,43 @@ public static (List<DeviceLaunchConfig> emulators, List<DeviceLaunchConfig> devi
             return (availableSimulators, availableDevices);
         }
 
-        public static void Build(string developmentTeamId)
+        public static void MenuBuild(string developmentTeamId)
         {
             try
             {
                 EditorUtility.DisplayProgressBar("Preparing your Mobile Client", "Building your XCode project", 0f);
-
-                if (!Directory.Exists(XCodeProjectPath))
-                {
-                    Debug.LogError("Unable to find an XCode project. Did you build your iOS worker?");
-                    return;
-                }
-
-                if (string.IsNullOrEmpty(developmentTeamId))
-                {
-                    Debug.LogError("Development Team Id was not specified. Unable to build the XCode project.");
-                    return;
-                }
-
-                if (!TryBuildXCodeProject(developmentTeamId, out var xcBuildErrors))
-                {
-                    Debug.LogError("Failed to build your XCode project. " +
-                        "Make sure you have the Command line tools for XCode (https://developer.apple.com/download/more/) " +
-                        $"installed and check the logs:\n{string.Join("\n", xcBuildErrors)}");
-                }
+                Build(developmentTeamId);
             }
             finally
             {
                 EditorUtility.ClearProgressBar();
             }
         }
 
+        public static void Build(string developmentTeamId = "")
+        {
+            if (!Directory.Exists(XCodeProjectPath))
+            {
+                throw new BuildFailedException(
+                    $"Unable to find an XCode project in {XCodeProjectPath}. " +
+                    "Did you build your iOS worker?");
+            }
+
+            if (string.IsNullOrEmpty(developmentTeamId) && IsXCodeProjectForDevice())
+            {
+                throw new BuildFailedException(
+                    "Development Team Id was not specified. Can't build this XCode project for device. " +
+                    "Either enter a Development Team Id or select the Simulator SDK as the target sdk.");
+            }
+
+            if (!TryBuildXCodeProject(developmentTeamId, out var xcBuildErrors))
+            {
+                throw new BuildFailedException("Failed to build your XCode project. " +
+					"Make sure you have the Command line tools for XCode (https://developer.apple.com/download/more/) " +
```

Comment:
change tabs to spaces
```suggestion
                    "Make sure you have the Command line tools for XCode (https://developer.apple.com/download/more/) " +
```